src/java/org/apache/cassandra/io/sstable/format/trieindex/TrieIndexSSTableWriter.java

```diff
@@ -310,9 +314,9 @@ class IndexWriter extends AbstractTransactional implements Transactional
         IndexWriter(TableMetadata table)
         {
             rowIndexFile = new SequentialWriter(descriptor.fileFor(Component.ROW_INDEX), WRITER_OPTION);
-            rowIndexFHBuilder = SSTableReaderBuilder.defaultIndexHandleBuilder(descriptor, Component.ROW_INDEX);
+            rowIndexFHBuilder = SSTableReaderBuilder.primaryIndexWriteTimeBuilder(descriptor, Component.ROW_INDEX, operationType);
```

Comment:
I think for consistency/safety (and because it's fairly trivial), we should also modify `BigTableWriter` similarly.